File: Python/8py 1/chromosome_check.py
"""
|--------------------------------------------------------------------------
| Determine offspring sex based on genes XX and XY chromosomes
|--------------------------------------------------------------------------
|
"""

def chromosome_check(sperm):
   return 'Congratulations! You\'re going to have a ' + ('son' if sperm == 'XY' else 'daughter') + '.'


def chromosome_check(sperm):
   return 'Congratulations! You\'re going to have a {}.'.format('son' if 'Y' in sperm else 'daughter')


def chromosome_check(sperm):
   return "Congratulations! You're going to have a %s." % ('son', 'daughter')[sperm[1] == 'X']


def chromosome_check(sperm):
   return "Congratulations! You're going to have a {}.".format({"XY": "son", "XX": "daughter"}.get(sperm)) 


print(chromosome_check('XY'), 'Congratulations! You\'re going to have a son.')
print(chromosome_check('XX'), 'Congratulations! You\'re going to have a daughter.')

File: JavaScript/8js 1/chromosomeCheck.js
/*
|--------------------------------------------------------------------------
| Determine offspring sex based on genes XX and XY chromosomes
|--------------------------------------------------------------------------
|
*/

function chromosomeCheck(sperm) {
   return sperm == "XY" ? "Congratulations! You're going to have a son." : "Congratulations! You're going to have a daughter."
}


function chromosomeCheck(sperm) {
   return `Congratulations! You're going to have a ${sperm == 'XY' ? 'son' : 'daughter'}.`
}


function chromosomeCheck(sperm) {
   return "Congratulations! You're going to have a " + (sperm == 'XY' ? 'son' : 'daughter') + '.';
}


function chromosomeCheck(sperm) {
   return `Congratulations! You're going to have a ${/Y/.test(sperm) ? 'son.' : 'daughter.'}`;
}


console.log(chromosomeCheck('XY'), "Congratulations! You're going to have a son.")
console.log(chromosomeCheck('XX'), "Congratulations! You're going to have a daughter.")